pkg/frontend/nvme.go

```diff
@@ -43,32 +43,32 @@ func calculateTransportAddr(pci *pb.PciEndpoint) string {
 // CreateNVMeController creates an NVMe controller
 func (s *Server) CreateNVMeController(ctx context.Context, in *pb.CreateNVMeControllerRequest) (*pb.NVMeController, error) {
 	log.Printf("Intel bridge CreateNVMeController received from client: %v", in.NvMeController)
-	if err := s.verifyNVMeController(in.NvMeController); err != nil {
+	if err := s.verifyNVMeControllerOnCreate(in.NvMeController); err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
 	// opi-spdk-bridge checks only IDs and returns success if a controller
 	// with such id exists ignoring any parameters. Check a controller
 	// exists with the same max/min limits
-	controller, ok := s.nvme.Controllers[in.NvMeController.Spec.Id.Value]
+	controller, ok := s.nvme.Controllers[in.NvMeControllerId]
```

Comment:
I think the unique key in the database is FULLNAME and not just ID...

Reply:
Yes, the database should contain a full name. However, at the moment, in opi-spdk-bridge an NVMe controller is saved under raw id specified at creation. Here, we need to check if we have this controller already created. `in.NvMeController.Spec.Id.Value` can be ignored by a user and only id is required to be provided. So I am using it to find a controller. When we have a function in opi-spdk-bridge responsible for transforming id to a full path, we can reuse it here... 

Or have I missed something?

Reply:
in https://github.com/opiproject/opi-spdk-bridge/blob/main/pkg/frontend/nvme.go#L270
we have
```
name = NvMeControllerId or randomUUID
in.NvMeController.Spec.Id = &pc.ObjectKey{Value: name}
```
so I think we should already use `in.NvMeController.Spec.Id`

Reply:
The problem with this chunk of code that a call to opi-spdk-bridge is not performed yet, the fullpath is not assigned/calculated, and we cannot rely on the value in in.NvMeController.Spec.Id here. Reorganized the code to avoid using fullpath until a call to opi-spdk-bridge is performed

Reply:
Yes, I agree. `Name` is better. We use ID or Handle in some cases... 
Since this is API change, I will start slack thread to unify and align all objects to have `name` first field...